### ISSUE TITLE
Cannot use import statement outside a module

### ISSUE THREAD

**Issue by koutilyagurjikindi** [callstack/react-native-paper#1993]
when I am running the npm test it showing this error in react native paper/react-native safe area view

### Current behaviour

● Test suite failed to run

node_modules/react-native-paper/node_modules/react-native-safe-area-view/index.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){import React, { Component } from 'react';
  

    SyntaxError: Cannot use import statement outside a module

      at ScriptTransformer._transformAndBuildScript (node_modules/@jest/transform/build/ScriptTransformer.js:537:17)
      at ScriptTransformer.transform (node_modules/@jest/transform/build/ScriptTransformer.js:579:25)
      at Object.<anonymous> (node_modules/react-native-paper/lib/commonjs/components/BottomNavigation.tsx:17:1)

Test Suites: 1 failed, 1 passed, 2 total 

### Code sample

jest file
module.exports = {
  // your Jest test options
  displayName: 'test',
  preset: 'react-native',
  transform: {
    '^.+\\.(js)$': require.resolve('react-native/jest/preprocessor.js'),
  },
  setupFiles: [
    '<rootDir>/node_modules/react-native-gesture-handler/jestSetup.js',
    '<rootDir>/node_modules/appcenter/test/AppCenterMock.js',
    '<rootDir>/node_modules/appcenter-analytics/test/AppCenterAnalyticsMock.js',
    '<rootDir>/node_modules/appcenter-crashes/test/AppCenterCrashesMock.js',
  ],
  testPathIgnorePatterns: ['<rootDir>/node_modules'],
  transformIgnorePatterns: [
    'node_modules/(?!(react-native' +
      '|react-native-cookies' +
      '|@sentry/react-native' +
      '|react-native-safe-area-view' +
      ')/)',
  ],
  collectCoverage: true,
};

### Your Environment

| software              | version
| --------------------- | -------
| ios or android        |
| react-native          |
| react-native-paper    |
| node                  |
| npm or yarn           |
| expo sdk              |

**Comment by koutilyagurjikindi** [comment]
Please help me to solve this error

**Comment by vamshi9666** [comment]
I had issues like this previously. Usually it was solved by mocking in a jest setup file

**Comment by DiegoBM** [comment]
Same issue here. The minute I include and start using react-native-paper, Jest will start throwing a similar error message to the one above

**Comment by rafaelbpa** [comment]
Any news about this? I'm with the same problem.

How this nice library doesn't work with jest? Or if does, there's no documentation about it. Anyone here can help?

**Comment by JossanPetrenko** [comment]
Same issue here. add react-native-paper, and jest stop

**Comment by rchaubey** [comment]
Spent like hours fixing this. Turned out my issue was using babel-jest instead of ts-jest.

[Follow this guide](https://kulshekhar.github.io/ts-jest/docs/guides/react-native/)

Make sure to change the `transform` property in `jest.config.js` to below

`  transform: {
    '^.+\\.(js)$': 'babel-jest',
    '\\.(ts|tsx)$': 'ts-jest',
  }`
  
Also add transformIgnorePatterns

`
  transformIgnorePatterns: [
    'node_modules/(?!react-native|react-native-paper)/',
  ],
`

Then I removed paper mocks from `__mocks__` and it worked.

Hopefully, it helps.
